tests/unit/lms/services/application_instance_test.py

```diff
@@ -75,20 +75,60 @@ def test_get_by_deployment_id_raises_on_missing(
         with pytest.raises(ApplicationInstanceNotFound):
             service.get_by_deployment_id(issuer, client_id, deployment_id)
 
-    @pytest.mark.parametrize(
-        "developer_secret,developer_key",
-        [
-            ("TEST_DEVELOPER_SECRET", "DEVELOPER_KEY"),
-            ("TEST_DEVELOPER_SECRET", None),
-            (None, "DEVELOPER_KEY"),
-            (None, None),
-        ],
-    )
+    @pytest.mark.parametrize("lms_url", (None, "http://lms-url.com"))
+    @pytest.mark.parametrize("deployment_id", (None, "DEPLOYMENT_ID"))
+    @pytest.mark.parametrize("developer_key", (None, "KEY"))
+    @pytest.mark.parametrize("developer_secret", (None, "SECRET"))
+    def test_update_application_instance(
+        self,
+        service,
+        application_instance,
+        lms_url,
+        deployment_id,
+        developer_key,
+        developer_secret,
+        aes_service,
+    ):
+
+        service.update_application_instance(
+            application_instance,
+            lms_url=lms_url,
+            deployment_id=deployment_id,
+            developer_key=developer_key,
+            developer_secret=developer_secret,
+        )
+
+        if developer_secret:
+            aes_service.build_iv.assert_called_once()
+            aes_service.encrypt.assert_called_once_with(
+                aes_service.build_iv.return_value, developer_secret
+            )
+
+            assert application_instance.developer_secret
+
+        if developer_key:
+            assert application_instance.developer_key == developer_key
```

Comment:
These tests don't make any assertions when these values are not truthy, so that side of the test doesn't test anything. We should probably make assertions about what should happen in both cases.

Reply:
> These tests don't make any assertions when these values are not truthy, so that side of the test doesn't test anything. We should probably make assertions about what should happen in both cases.

Not ignoring this feedback, handling it in a different PR, want to get merge all the admin PRs at the same time.